src/prefect/new_task_engine.py

```diff
@@ -460,7 +465,7 @@ async def run_task_async(
     )
 
     # This is a context manager that keeps track of the run of the task run.
-    with engine.start(task_run_id=task_run_id) as run:
+    with engine.start(task_run_id=task_run_id, dependencies=dependencies) as run:
         run.begin_run()
```

Comment:
Okay this is a subtlety I think I missed earlier, so there's an engine for _each_ task run, it's not a long-lived engine that could run many of them.  :muscle: